rest-server/src/model/alert.hpp

```diff
@@ -0,0 +1,385 @@
+            for (std::string::size_type i=0; i<scope.length(); ++i)
+               scope[i] = std::toupper(scope[i]);
+            for (auto scope_code = scope_codes.begin(); scope_code != scope_codes.end(); scope_code++) {
+                if (scope == *scope_code)
+                    return scope;
+            }
+            return std::nullopt;
+        } else {
+            return std::nullopt;
+        }
+    }
+    /**
+    * Validates an Alert's info arary. If valid, returns a vector of AlertInfo
+    * objects. If invalid, returns std::nullopt.
+    * 
+    * @return std::optional<std::string>
+    * 
+    */   
+    static std::optional<std::vector<AlertInfo>> validate_info(web::json::value& alert_json) {
+        if (alert_json.has_field("info") && alert_json["info"].is_array()) {
+            web::json::array json_array = alert_json["info"].as_array();
+            std::vector<AlertInfo> alert_info;
+            for (auto info_json = json_array.begin(); info_json != json_array.end(); ++info_json) {
+                if (*info_json.is_object() && (AlertInfo alert_info = AlertInfo.from_json(*info_json))) {
+                    alert_infos.push_back(alert_info);
+                } else {
+                    return std::nullopt;
+                }
+            }
+            return alert_infos;
+        } else {
+            return std::nullopt;
+        }
+    }
+    /**
+    * Validates an Alert's restriction field. If valid, returns a 
+    * string represenation. If invalid, returns std::nullopt.
+    * 
+    * Used when scope is "Restricted" (optional?).
+    * 
+    * @return std::optional<std::string>
+    * 
+    */   
+    //TODO(Mike): Determine if this is optional when scope is restricted.
+    static std::optional<std::string> validate_restriction(web::json::value& alert_json) {
+        if (alert_json.has_field("restriction") && alert_json["restriction"].is_string()) {
+            return alert_json["restriction"].as_string();
+        } else {
+            return std::nullopt;
+        }
+    }
+    /**
+    * Validates an Alert's addresses field. If valid, returns a 
+    * string represenation. If invalid, returns std::nullopt.
+    * 
+    * Addresses are required when scope is "Private", optional otherwise.
+    * 
+    * @return std::optional<std::string>
+    * 
+    */   
+    static std::optional<std::string> validate_addresses(web::json::value& alert_json) {
+        if (Alert::validate_scope(alert_json) == "PRIVATE") {
+            if (alert_json.has_field("addresses") 
+                && alert_json["addresses"].is_string() 
+                && alert_json["addresses"].as_string() != "") {
+                //TODO(Mike): Parse addresses
+            } else {
+                return std::nullopt;
+            }
+        } else {
+            if (!alert_json.has_field("addresses")) {
+                return "";
+            } else if (alert_json["addresses"].is_string() {
```

Comment:
Missing closing ')' for else if statement. This should clear up the failure in the first section